### PR TITLE
[NGRM] There should be an option to use constant color when coloring …

### PR DESCRIPTION
**Pull request by nilscb**
…the MapLayer #1414

### CODE REVIEW

**Comment by hkfb** [comment]
Not a fan. It's easy to make a callback that always returns the same color. To make it even easier we could make a utility hook that always returns a specified color.

**Comment by nilscb** [comment]
Yes I agree. There is already the colormap function property and setting that to return a constant color should do the trick. There was a problem however to make this work partly due to the redux  but since redux is removed now Ill try again and abandon this pull request.

**Comment by t0oF-azpn** [comment]
What about performance or work load ?
For big data, there can be an impact of doing per vertex color extraction and color OpenGL calls. It's kind of strange to send a color array along with the vertex array down the OpenGL pipeline while it is not needed at all...

**Comment by nilscb** [comment]
Thats a valid point I belive. Maybe keep the API (ie setting colomap funtion to return constant color) But in this case internally just use an unifirm for the color.

**Comment by hkfb** [comment]
> Thats a valid point I belive. Maybe keep the API (ie setting colomap funtion to return constant color) But in this case internally just use an unifirm for the color.

The `colorMap` prop could be optionally a single `[r, g, b]` instead of introducing a new prop. I think this is a common "Accessor" pattern elsewhere in Deck.gl

**Comment by nilscb** [comment]
I'm redefining colorMapFunction to be either a function or a color. Then if just a color I will use a uniform.

File: react/src/lib/components/SubsurfaceViewer/layers/map/mapLayer.stories.tsx [hunk 1]
```diff
@@ -279,6 +279,44 @@ MapLayer3dPng.parameters = {
...
+            },
+        ],
+    },
+};
+
+MapLayerConstantColor.parameters = {
+    docs: {
+        ...defaultParameters.docs,
+        description: {
+            story: 'Example using the property"useConstantColor" to color the surface in one color only',
```
Comment (hkfb) [inline react/src/lib/components/SubsurfaceViewer/layers/map/mapLayer.stories.tsx:315]: Should update this text.

**Review by hkfb — CHANGES_REQUESTED** [review]

**Review by hkfb — APPROVED** [review]

File: react/src/lib/components/SubsurfaceViewer/layers/map/mapLayer.ts [hunk 2]
```diff
@@ -319,12 +321,19 @@ export default class MapLayer extends CompositeLayer<MapLayerProps<unknown>> {
...
                     e.data;
 
+                const isConstantColor =
+                    typeof this.props.colorMapFunction !== "function";
+
+                const colorFunc = !isConstantColor
+                    ? () => this.props.colorMapFunction
+                    : this.props.colorMapFunction;
+
                 const legend = {
```
[the remark this answers is not in the document]
Comment (nilscb) [reply react/src/lib/components/SubsurfaceViewer/layers/map/mapLayer.ts]: Will fix this

File: react/src/lib/components/SubsurfaceViewer/layers/map/mapLayer.stories.tsx [hunk 3]
```diff
@@ -279,6 +279,44 @@ MapLayer3dPng.parameters = {
     },
 };
 
+export const MapLayerConstantColor: ComponentStory<typeof SubsurfaceViewer> = (
```
[the remark this answers is not in the document]
Comment (nilscb) [reply react/src/lib/components/SubsurfaceViewer/layers/map/mapLayer.stories.tsx]: Will fix this

**Pull request merged**